src/app/hooks/useDistributionStore.ts

```diff
@@ -0,0 +1,20 @@
+import { NETWORKS } from '~lib/utils/envHelper';
+import { useStores } from '~app/hooks/useStores';
+import { DEVELOPER_FLAGS, getLocalStorageFlagValue } from '~lib/utils/developerHelper';
+import DistributionStore from '~app/common/stores/applications/Distribution/Distribution.store';
+import DistributionTestnetStore from '~app/common/stores/applications/Distribution/DistributionTestnet.store';
+
+export const useDistributionStore = (networkId: number) => {
+  const stores = useStores();
+  const mainnetStore: DistributionStore = stores.Distribution;
+  const testnetStore: DistributionTestnetStore = stores.DistributionTestnet;
+  const enableTestingFlow = getLocalStorageFlagValue(DEVELOPER_FLAGS.ENABLE_TEST_DISTRIBUTION_FLOW);
+  const currentFlow = enableTestingFlow ? enableTestingFlow : networkId;
+
+  const distributionStores = {
+    [NETWORKS.GOERLI]: testnetStore,
+    [NETWORKS.MAINNET]: mainnetStore,
+  };
```

Comment:
Well, i see mainnet and goerly (old ot new?)
We should have support for old goerli and new maknnet. 

But in addition to that, we should have ABI equal to mainnet to work with for new goerli distribution (for testing purposes). 

Please explain how to achieve this. 
In your task please

Reply:
we have additional distributionHelper, which return contract and ABI by network ID, but, we also have there some logic which check future flag for testing, and if he exist he return the abi which equal to mainnet and also contract addres which we will use to testing, contract address for testing also should be provide in local storage